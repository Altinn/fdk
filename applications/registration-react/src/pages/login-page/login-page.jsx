import pick from 'lodash/pick';
import React from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';
import { withProps } from 'recompose';

import localization from '../../services/localization';
import { login } from '../../services/auth/auth-service';
import './login-page.scss';
import { getLoginState } from '../../services/auth/login-store';

const showLoginReadOnly = window.localStorage.getItem('showLoginReadOnly');

const renderMessageForReauthenticateDueToTimeout = () => (
  <div className="row">
    <div className="col-12">
      <div
        className="mt-2 alert alert-warning fdk-color-neutral-darkest"
        role="alert"
      >
        <span>
          <strong>{localization.loginDialog.loggedOutMsgPart1}</strong>
        </span>
        <span>{localization.loginDialog.loggedOutMsgPart2}</span>
      </div>
    </div>
  </div>
);

export const LoginPagePure = ({ reauthenticateDueToTimeout }) => (
  <>
    <div className="login-dialog-wrapper pt-5">
      <div className="container">
        {reauthenticateDueToTimeout &&
          renderMessageForReauthenticateDueToTimeout()}
        <div className="row">
          {showLoginReadOnly && (
            <div className="col-md-6">
              <div className="jumbotron jumbotron-login h-100">
                <h2>{localization.app.loginPage.readUser.heading}</h2>
                <Button
                  className="fdk-button fdk-button-cta fdk-bg-color-link mt-4"
                  onClick={() => login({ readOnly: true })}
                >
                  {localization.app.loginPage.readUser.loginButton}
                </Button>
                <p>{localization.app.loginPage.readUser.description1}</p>
                <div>
                  <strong>
                    {localization.app.loginPage.readUser.description2}
                  </strong>
                  <p>
                    {localization.app.loginPage.readUser.description3.click}{' '}
                    <strong>
                      {localization.app.loginPage.readUser.loginButton}
                    </strong>
                    , {localization.app.loginPage.readUser.description3.then}{' '}
                    <strong>
                      {
                        localization.app.loginPage.readUser.description3
                          .registerUser
                      }
                    </strong>
                    .
                  </p>
                </div>
              </div>
            </div>
          )}
          <div className="col-md-6 mt-5 mt-md-0">
            <div className="jumbotron jumbotron-login h-100">
              <h2>{localization.app.loginPage.writeUser.heading}</h2>
              <Button
                className="fdk-button fdk-button-cta fdk-bg-color-link mt-4"
                onClick={login}
              >
                {localization.app.loginPage.writeUser.loginButton}
              </Button>
              <p className="">
                {localization.app.loginPage.writeUser.description1}
              </p>
              <div>
                <strong>
                  {localization.app.loginPage.writeUser.description2}
                </strong>
                <p>
                  <a href="https://fellesdatakatalog.brreg.no/about-registration">
                    {localization.app.loginPage.writeUser.description3}
                  </a>
                </p>
                <p />
              </div>
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-12 ml-4 mt-4">
            <div>
              <strong>{localization.app.loginPage.postamble1}</strong>
              <p>
                {localization.app.loginPage.postamble2}
                <br />
                {localization.app.loginPage.postamble3}{' '}
                <a href="https://fellesdatakatalog.brreg.no/about-registration">
                  {localization.app.loginPage.postamble4}
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </>
);

LoginPagePure.defaultProps = {
  reauthenticateDueToTimeout: false
};

LoginPagePure.propTypes = {
  reauthenticateDueToTimeout: PropTypes.bool
};

const withLoginProps = withProps(() =>
  pick(getLoginState(), 'reauthenticateDueToTimeout')
);

export const LoginPage = withLoginProps(LoginPagePure);
