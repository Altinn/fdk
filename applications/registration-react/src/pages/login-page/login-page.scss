@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.login-dialog-wrapper {
  background-color: $color-primary-lighter;
  margin-top: -3rem;
  margin-bottom: -6rem;
  flex-grow: 1;
  .fdk-button {
    text-decoration: none;
    margin-bottom: 1em;
    border-color: transparent;
    &:hover {
      color: $color-white;
      border-color: transparent;
    }
  }
}

/*==========  Media Queries  ==========*/
/* Custom, iPhone Retina */
@media only screen and (min-width : 320px) {

}/* end min-width : 320px */


/* Extra Small Devices, Phones */
@media only screen and (min-width : 480px) {

}/* end min-width : 480px */

/* Small Devices, Tablets */
@media only screen and (min-width: 768px) {
  .login-dialog-wrapper {
    background-image: url(../../assets/img/illustration-login.svg);
    background-repeat: no-repeat;
    background-position: top right;
    background-size: 50%;
  }
}

/* Medium Devices, Desktops */
@media only screen and (min-width : 992px) {
  .login-dialog-wrapper {
    background-size: contain;
  }
}

/* Large Devices, Wide Screens */
@media only screen and (min-width : 1200px) {
}

.jumbotron-login {
  box-shadow: 0 3px 12px rgba(0, 0, 0, 0.15);
  background-color: white !important;
  padding: 2rem 2rem 0rem 2rem !important;
}
