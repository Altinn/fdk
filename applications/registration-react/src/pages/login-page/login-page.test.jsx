import React from 'react';
import { shallow } from 'enzyme';
import { LoginPagePure } from './login-page';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    loggedOut: true
  };
  wrapper = shallow(<LoginPagePure {...defaultProps} />);
});

test('should render LoginPage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
