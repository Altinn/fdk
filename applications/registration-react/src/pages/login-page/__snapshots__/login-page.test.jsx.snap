// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<Fragment>
  <div
    className="login-dialog-wrapper pt-5"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="mt-2 alert alert-warning fdk-color-neutral-darkest"
            role="alert"
          >
            <span>
              <strong>
                Du har blitt logget ut. 
              </strong>
            </span>
            <span>
              Sesjonen ble tidsavbrutt, men vi har tatt vare på alle endringene du har gjort. Logg inn på nytt for å fortsette å jobbe med registreringen.
            </span>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-6 mt-5 mt-md-0"
        >
          <div
            className="jumbotron jumbotron-login h-100"
          >
            <h2>
              For deg som har skrivetilgang
            </h2>
            <Button
              className="fdk-button fdk-button-cta fdk-bg-color-link mt-4"
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              Logg inn med ID porten
            </Button>
            <p
              className=""
            >
              For å få tilgang til funksjonalitet som registrering og redigering på vegne av din virksomhet, må du logge inn med IP porten.
            </p>
            <div>
              <strong>
                Trenger du skrivetilgang?
              </strong>
              <p>
                <a
                  href="https://fellesdatakatalog.brreg.no/about-registration"
                >
                  Les hvordan du får tilgang til å registrere og redigere innhold.
                </a>
              </p>
              <p />
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12 ml-4 mt-4"
        >
          <div>
            <strong>
              Skal du gi andre skrivetilgang?
            </strong>
            <p>
              Du kan gi andre til a redigere datasett for din virksomhet via Altinn.
              <br />
              Les mer om
               
              <a
                href="https://fellesdatakatalog.brreg.no/about-registration"
              >
                hvordan du gir andre tilgang.
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
